packages/daemon/src/host.js

```diff
@@ -432,6 +436,84 @@ export const makeHostMaker = ({
       return value;
     };
 
+    /**
+     * @param {string} guestName
+     */
+    const invite = async guestName => {
+      // We must immediately retain a formula under guestName so that we
+      // preserve the invitation across restarts, but we must replace the
+      // guestName with the handle of the guest that accepts the invitation.
+      // We need to return the locator for the invitation regardless of what
+      // we store.
+      // Overwriting the guestName must cancel the pending invitation (consume
+      // once) so that the invitation can no longer modify the petStore entry
+      // for the guestName.
+      /** @type {import('./types.js').DeferredTasks<import('./types.js').InvitationDeferredTaskParams>} */
+      const tasks = makeDeferredTasks();
+      tasks.push(identifiers =>
+        petStore.write(guestName, identifiers.invitationId),
+      );
+      const { value } = await formulateInvitation(
+        hostId,
+        handleId,
+        guestName,
+        tasks,
+      );
+      return value;
+    };
+
+    /**
+     * @param {string} invitationLocator
+     * @param {string} guestName
+     */
+    const accept = async (invitationLocator, guestName) => {
+      const url = new URL(invitationLocator);
+      const nodeNumber = url.hostname;
+      const invitationNumber = url.searchParams.get('id');
+      const remoteHandleNumber = url.searchParams.get('from');
+      const addresses = url.searchParams.getAll('at');
```

Comment:
Should we also validate this? All other extracted URL components appear to be validated. At present, the ultimately thrown error is "no supported networks", which is not very helpful if `addresses` were somehow empty.

Reply:
I think this code needs to be very temporary. A lot of this should be done automatically by lower-level mechanisms. Eg., `E(hub).write(name, locator)` should extract `{ node, formula, type, addresses }` from the locator, submit `{ node, addresses }` to the known peers store, capture `{node, formula, type }` in the pet store, and likewise `provide` should operate on only `{ node, formula }`.